include/rocksdb/cache_type.h

```diff
@@ -0,0 +1,39 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
```

Comment:
I created a separate file for cache types and conversions because they will soon be used in cache bench and db bench.

Reply:
`include/rocksdb` is the public API. FastLRUCache is not intended to be exposed in the public API (at least not yet) and an enum like this in the public API runs counter to the idea of Cache being Customizable (eventually): that custom implementations are, as much as possible, given equal treatment to built-in implementations.

Even if we make this an internal-only enum, I don't see what an enum gets us if we're always just doing string -> if/else -> enum -> switch/if-else on that enum, when we could just do if-else on the string.

It looks like we already have a Cache::CreateFromString function that I think we should be using here, except we need the `id=` feature from #10084. Let me see if I can fast-track #10084 to untangle the parallel development mess here.

Reply:
@pdillinger Are you suggesting waiting for #10084 before merging this PR, or change this PR to use string directly instead of enum and use Cache::CreateFromString later when that's ready?

Reply:
@anand1976 I talked to Peter and we agreed to do the latter.